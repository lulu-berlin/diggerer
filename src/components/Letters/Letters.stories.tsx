import React from "react";
import { storiesOf } from "@storybook/react";
import { LettersSymbols } from "components/Letters/LettersSymbols";
import { Letters } from "components/Letters/Letters";

console.log(module);

storiesOf("Letters", module)
  .add("hello world", () => (
    <>
      <LettersSymbols />
      <Letters charWidth={16} charHeight={16}>
        hello world
      </Letters>
    </>
  ))
  .add("DIGGER", () => (
    <>
      <LettersSymbols />
      <Letters charWidth={16} charHeight={16}>
        DIGGER
      </Letters>
    </>
  ))
  .add("HIGH SCORE", () => (
    <>
      <LettersSymbols />
      <Letters charWidth={16} charHeight={16}>
        HIGH SCORE
      </Letters>
    </>
  ));
