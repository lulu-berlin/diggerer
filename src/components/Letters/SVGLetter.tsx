import React from "react";

export type LetterProps = React.SVGProps<SVGUseElement> & {
  /**
   * The letter to render (a-z or A-Z).
   */
  char: string;
};

/**
 * This component renders a letter from the SVG symbols created by <LettersSymbols>.
 */
export const SVGLetter: React.FC<LetterProps> = ({ char, ...svgProps }) => {
  const letter = char.toUpperCase();

  return /[A-Z]/.test(letter) ? (
    <use href={`#CLET${char.toUpperCase()}`} {...svgProps} />
  ) : null;
};
