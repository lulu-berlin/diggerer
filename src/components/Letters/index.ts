export * from "./LettersSymbols";
export * from "./Letters";
