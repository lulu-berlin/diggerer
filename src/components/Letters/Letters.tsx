import React from "react";
import { SVGLetter } from "components/Letters/SVGLetter";

export type LettersProps = {
  children: string;
  charWidth: number;
  charHeight: number;
};

export const Letters: React.FC<LettersProps> = ({
  children: text,
  charWidth,
  charHeight
}) => (
  <svg height={charHeight} width={charWidth * text.length}>
    {text.split("").map((char, i) => (
      <SVGLetter
        char={char}
        x={charWidth * i}
        y={0}
        width={charWidth}
        height={charHeight}
        key={i}
      />
    ))}
  </svg>
);
