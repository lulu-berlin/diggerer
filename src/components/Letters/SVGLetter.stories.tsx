import React from "react";
import { storiesOf } from "@storybook/react";
import { LettersSymbols } from "components/Letters/LettersSymbols";
import { SVGLetter } from "components/Letters/SVGLetter";

const A = "A".charCodeAt(0);
const Z = "Z".charCodeAt(0);

Array.from({ length: Z - A + 1 }).forEach((_, i) => {
  const char = String.fromCharCode(A + i);

  storiesOf("SVGLetter", module).add(char, () => (
    <>
      <LettersSymbols />
      <svg width={16} height={16}>
        <SVGLetter x={0} y={0} width={16} height={16} char={char} />
      </svg>
    </>
  ));
});
