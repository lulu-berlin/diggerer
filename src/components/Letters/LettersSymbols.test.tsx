import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";
import { LettersSymbols } from "components/Letters";

it("should render LettersSymbols correctly", () => {
  const wrapper = shallow(<LettersSymbols />);
  expect(wrapper).toMatchSnapshot();
});
