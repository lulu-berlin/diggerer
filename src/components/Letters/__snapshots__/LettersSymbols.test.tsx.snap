// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LettersSymbols correctly 1`] = `
<styled.svg
  style={
    Object {
      "display": "none",
    }
  }
>
  <symbol
    id="CLETA"
    viewBox="0 0 12 12"
  >
    <path
      d="M1.6.8c-.3.5-.8 3.2-1.2 6-.6 4.4-.4 5.2 1 5.2 1 0 1.6-.9 1.6-2.5C3 7.5 3.5 7 5.5 7S8 7.5 8 9.5c0 1.4.5 2.5 1.1 2.5.8 0 .9-1.6.5-5.2C8.7.3 8.5 0 5 0 3.4 0 1.9.4 1.6.8zM7 3.5C7 4.3 6.1 5 5 5s-2-.7-2-1.5S3.9 2 5 2s2 .7 2 1.5z"
    />
  </symbol>
  <symbol
    id="CLETB"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.3C.1 2.1 0 4.7.2 7.1l.3 4.4H5c4.2 0 4.5-.2 4.8-2.8C10 7.2 9.7 6 9.1 6 8.6 6 8 4.8 7.8 3.2 7.6 1 7 .4 4.2.2 2.1 0 .8.4.4 1.3zM6 3.5C6 4.3 5.1 5 4 5s-2-.7-2-1.5S2.9 2 4 2s2 .7 2 1.5zm2 5c0 .9-.9 1.5-2.5 1.5S3 9.4 3 8.5 3.9 7 5.5 7 8 7.6 8 8.5z"
    />
  </symbol>
  <symbol
    id="CLETC"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.4C.1 2.1 0 4.7.2 7.1l.3 4.4 4.3.3c3.1.2 4.4-.1 4.8-1.3.4-.9.3-1.9-.1-2.3-.4-.4-1-.1-1.2.5-.7 2-5.3 1.6-5.3-.5 0-.9-.3-2.7-.6-4C1.8 2.3 2.1 2 4.8 2c1.7 0 3.2.6 3.5 1.2.2.7.8 1 1.2.6.4-.4.5-1.4.2-2.3-.8-2-8.5-2.1-9.3-.1z"
    />
  </symbol>
  <symbol
    id="CLETD"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 6v6.1l4.8-.3 4.7-.3V.5L4.8.2 0-.1V6zm8 0c0 3.3-.3 4-2 4-1.3 0-2-.7-2-2 0-1.1-.4-2-1-2-.5 0-1-.9-1-2 0-1.6.7-2 3-2 2.8 0 3 .2 3 4z"
    />
  </symbol>
  <symbol
    id="CLETE"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.4C.1 2.1 0 4.7.2 7.1l.3 4.4 4.8.3c2.8.2 4.7-.1 4.7-.7S8.7 10 7 10c-2 0-3-.5-3-1.5C4 7.7 4.9 7 6 7s2-.5 2-1c0-.6-1.3-1-3-1-2 0-3-.5-3-1.5C2 2.4 3.1 2 6 2c2.2 0 4-.5 4-1C10-.6 1-.2.4 1.4z"
    />
  </symbol>
  <symbol
    id="CLETF"
    viewBox="0 0 12 12"
  >
    <path
      d="M.7.7c-.4.3-.7 3-.7 6C0 11.3.3 12 2 12c1.5 0 2-.7 2-2.5S4.5 7 6 7c1.1 0 2-.5 2-1 0-.6-1.1-1-2.5-1C3.9 5 3 4.4 3 3.5 3 2.5 4.1 2 6.5 2 8.4 2 10 1.5 10 1 10-.1 1.7-.4.7.7z"
    />
  </symbol>
  <symbol
    id="CLETG"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.4C.1 2.1 0 4.7.2 7.1l.3 4.4H5c3.9 0 4.5-.3 4.8-2.3C10.1 7 7.6 5 6.2 6.5c-.4.4-.1 1 .6 1.2 1.8.7 1.5 2.3-.5 2.3C4.6 10 2 6 2 3.3c0-1.7 5.7-1.7 6.3-.1.2.7.8 1 1.2.6.4-.4.5-1.4.2-2.3-.8-2-8.5-2.1-9.3-.1z"
    />
  </symbol>
  <symbol
    id="CLETH"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 6c0 5.3.2 6 2 6 1.5 0 2-.7 2-2.5S4.5 7 6 7s2 .7 2 2.5c0 1.4.5 2.5 1 2.5.6 0 1-2.7 1-6s-.4-6-1-6c-.5 0-1 1.1-1 2.5C8 4.6 7.5 5 5 5s-3-.4-3-2.5C2 1.1 1.6 0 1 0S0 2.7 0 6z"
    />
  </symbol>
  <symbol
    id="CLETI"
    viewBox="0 0 12 12"
  >
    <path
      d="M4 6c0 5.3.2 6 2 6 1.6 0 2-.7 2-3 0-1.7-.4-3-1-3-.5 0-1-1.4-1-3 0-1.7-.4-3-1-3S4 2.7 4 6z"
    />
  </symbol>
  <symbol
    id="CLETJ"
    viewBox="0 0 12 12"
  >
    <path
      d="M6 5c0 4.3-.3 5-2 5-1.1 0-2-.7-2-1.5S1.5 7 .9 7C.3 7 0 8 .2 9.2c.3 2 .9 2.3 4.8 2.3 4.3 0 4.5-.1 4.8-3.3C10 6.4 9.7 5 9.1 5S8 3.9 8 2.5 7.6 0 7 0c-.5 0-1 2.2-1 5z"
    />
  </symbol>
  <symbol
    id="CLETK"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 6c0 5.3.2 6 2 6 1.5 0 2-.7 2-2.5S4.5 7 6 7s2 .7 2 2.5c0 1.4.5 2.5 1 2.5.6 0 1-1.3 1-3s-.4-3-1-3c-.5 0-1-1.4-1-3 0-1.7-.4-3-1-3-.5 0-1 1.1-1 2.5C6 4.3 5.5 5 4 5s-2-.7-2-2.5C2 1.1 1.6 0 1 0S0 2.7 0 6z"
    />
  </symbol>
  <symbol
    id="CLETL"
    viewBox="0 0 12 12"
  >
    <path
      d="M.2 5.7l.3 5.8 4.8.3c2.8.2 4.7-.1 4.7-.7S8.7 10 7 10c-2.5 0-3-.4-3-2.5C4 6.1 3.6 5 3 5c-.5 0-1-1.1-1-2.5S1.5 0 1 0C.3 0 0 2.2.2 5.7z"
    />
  </symbol>
  <symbol
    id="CLETM"
    viewBox="0 0 12 12"
  >
    <path
      d="M.7.6C.3 1 0 3.7 0 6.7 0 10.9.3 12 1.6 12c1.4 0 1.6-.8 1-5-.4-3.2-.3-5 .4-5 .5 0 1 2.2 1 5 0 2.7.5 5 1 5 .6 0 1-2.3 1-5 0-2.8.5-5 1-5 .6 0 1 2.2 1 5s.5 5 1 5c.7 0 1-2.2.8-5.7L9.5.5l-4-.3C3.2 0 1.1.2.7.6z"
    />
  </symbol>
  <symbol
    id="CLETN"
    viewBox="0 0 12 12"
  >
    <path
      d="M1.7.6C1.3 1 1 3.7 1 6.7c0 4.2.3 5.3 1.6 5.3 1.4 0 1.6-.8 1-5-.7-4.9-.6-5 1.9-5C7.8 2 8 2.3 8 7c0 2.8.5 5 1 5 .7 0 1-2.2.8-5.7L9.5.5 6 .2C4 0 2.1.2 1.7.6z"
    />
  </symbol>
  <symbol
    id="CLETO"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.3C.1 2.1 0 4.7.2 7.1l.3 4.4h9V.5L5.2.2C2.2 0 .8.3.4 1.3zm6.2 2.1C7.4 4.2 8 6 8 7.4 8 9.6 7.6 10 5 10c-2.8 0-3-.2-3-4C2 1.7 3.8.7 6.6 3.4z"
    />
  </symbol>
  <symbol
    id="CLETP"
    viewBox="0 0 12 12"
  >
    <path
      d="M.7.6C.3 1 0 3.7 0 6.7 0 11.3.3 12 2 12c1.5 0 2-.7 2-2.4 0-1.9.6-2.5 2.8-2.8 2.3-.3 2.7-.8 2.7-3.3 0-2.8-.3-3-4-3.3C3.2 0 1.1.2.7.6zM8 3.5C8 4.5 7 5 5 5s-3-.5-3-1.5S3 2 5 2s3 .5 3 1.5z"
    />
  </symbol>
  <symbol
    id="CLETQ"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.3C.1 2.1 0 4.7.2 7.1l.3 4.4h9V.5L5.2.2C2.2 0 .8.3.4 1.3zM8 5c0 2-.5 3-1.5 3C5.7 8 5 8.4 5 9c0 .5-.7 1-1.5 1C2.4 10 2 8.9 2 6c0-3.8.2-4 3-4 2.7 0 3 .3 3 3z"
    />
  </symbol>
  <symbol
    id="CLETR"
    viewBox="0 0 12 12"
  >
    <path
      d="M.7.6C-.4 1.7-.1 12 1 12c.6 0 1-1.1 1-2.5C2 7.4 2.5 7 5 7s3 .4 3 2.5c0 1.4.5 2.5 1 2.5.6 0 1-1.3 1-3s-.4-3-.9-3S8 4.8 7.8 3.2C7.6 1 6.9.4 4.5.2 2.8 0 1.1.2.7.6zM6 3.5C6 4.3 5.1 5 4 5s-2-.7-2-1.5S2.9 2 4 2s2 .7 2 1.5z"
    />
  </symbol>
  <symbol
    id="CLETS"
    viewBox="0 0 12 12"
  >
    <path
      d="M.4 1.4C-.7 4.3.5 6.5 3.3 6.8 4.9 7 6 7.7 6 8.7c0 1.2-.7 1.4-3.1.9-2.2-.4-2.9-.2-2.6.7.2.7 2.2 1.2 4.8 1.2 4.2 0 4.4-.1 4.4-3 0-2.7-.3-3-3.7-3.3C3.1 5 2 4.4 2 3.3c0-1.3.8-1.5 4.1-1 2.8.4 3.9.2 3.6-.6C9.1-.2 1.1-.4.4 1.4z"
    />
  </symbol>
  <symbol
    id="CLETT"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 1c0 .5.9 1 2 1 1.7 0 2 .7 2 5s.3 5 2 5c1.6 0 2-.7 2-3 0-1.7-.4-3-1-3-.5 0-1-.9-1-2 0-1.3.7-2 2-2 1.1 0 2-.5 2-1 0-.6-2.2-1-5-1-2.7 0-5 .4-5 1z"
    />
  </symbol>
  <symbol
    id="CLETU"
    viewBox="0 0 12 12"
  >
    <path
      d="M.2 5.7l.3 5.8h9l.3-5.8C10 2.2 9.7 0 9 0c-.5 0-1 2.2-1 5 0 4.3-.3 5-2 5-1.5 0-2-.7-2-2.5C4 6.1 3.6 5 3 5c-.5 0-1-1.1-1-2.5S1.5 0 1 0C.3 0 0 2.2.2 5.7z"
    />
  </symbol>
  <symbol
    id="CLETV"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 3.4C0 9 1.6 12 4.6 12c1.7 0 2.8-.6 3.1-2 .3-1.1.9-2 1.4-2 .5 0 .9-1.8.9-4s-.4-4-1-4c-.5 0-1 1.6-1 3.6 0 5-3.9 8.6-5.4 4.8-.3-.9-.6-3.1-.6-5S1.6 0 1 0C.5 0 0 1.5 0 3.4z"
    />
  </symbol>
  <symbol
    id="CLETW"
    viewBox="0 0 12 12"
  >
    <path
      d="M.2 5.7l.3 5.8h9l.3-5.8C10 2.2 9.7 0 9 0c-.5 0-1 2.2-1 5 0 2.7-.4 5-1 5-.5 0-1-2.3-1-5 0-6.6-1.8-6.4-2.2.2l-.4 5.3-.8-5.3c-1.1-7-2.8-6.6-2.4.5z"
    />
  </symbol>
  <symbol
    id="CLETX"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 2.4c0 1.3.6 2.7 1.3 3 1 .5.9.7 0 1.2-.7.3-1.3 1.7-1.3 3C0 11.3.6 12 2 12c1.5 0 2-.7 2-2.5S4.5 7 6 7s2 .7 2 2.5c0 1.4.5 2.5 1 2.5 1.4 0 1.2-4.7-.2-5.4-1-.5-1-.7 0-1.2C10.2 4.7 10.4 0 9 0c-.5 0-1 1.1-1 2.5C8 4.6 7.5 5 5 5s-3-.4-3-2.5C2 1.1 1.6 0 1 0 .5 0 0 1.1 0 2.4z"
    />
  </symbol>
  <symbol
    id="CLETY"
    viewBox="0 0 12 12"
  >
    <path
      d="M0 3.1c0 2.2.5 3.2 2 3.6 1.3.3 2 1.4 2 2.9S4.6 12 5.5 12 7 11.1 7 9.6c0-1.3.7-2.6 1.5-3C10.2 6 10.6 0 9 0c-.5 0-1 1.1-1 2.5C8 4.6 7.5 5 5 5s-3-.4-3-2.5C2 1.1 1.6 0 1 0 .5 0 0 1.4 0 3.1z"
    />
  </symbol>
  <symbol
    id="CLETZ"
    viewBox="0 0 12 12"
  >
    <path
      d="M.7.6C-.9 2.2.2 2.8 4 2.3c3.2-.5 4-.2 4 1C8 4.4 6.9 5 4.3 5.2.8 5.5.5 5.8.5 8.5c0 2.9.2 3 4.4 3 2.6 0 4.6-.5 4.8-1.2.3-.9-.4-1.1-2.6-.7-2.4.5-3.1.3-3.1-.9 0-1 1.1-1.7 2.8-1.9 2.3-.3 2.7-.8 2.7-3.3 0-2.8-.3-3-4-3.3C3.2 0 1.1.2.7.6z"
    />
  </symbol>
</styled.svg>
`;
